config/src/main/java/org/springframework/security/config/annotation/web/configurers/saml2/Saml2LoginConfigurer.java

```diff
@@ -335,6 +339,16 @@ private Saml2AuthenticationRequestFactory getResolver(B http) {
 			}
 			return resolver;
 		}
+
+		private Saml2AuthenticationRequestContextResolver getContextResolver(B http) {
+			Saml2AuthenticationRequestContextResolver resolver = getBeanOrNull(http, Saml2AuthenticationRequestContextResolver.class);
+			if (resolver == null) {
+				return new DefaultSaml2AuthenticationRequestContextResolver(
```

Comment:
I think it would be better to have a single `return` statement.
`new DefaultSaml2AuthenticationRequestContextResolver` can be moved into `getDefaultContextResolver` method.

Reply:
I'm of a different opinion on this one. I think returning early is more readable. 

However, since that's probably in the eye of the beholder, what I'd propose for now is aligning with how other similar methods in the class, like `getResolver`. We can consider changing both methods at another time.